internal/services/cdn/cdn_frontdoor_helpers.go

```diff
@@ -200,10 +208,31 @@ func expandStringSliceToCsvFormat(input []interface{}) *string {
 	return &csv
 }
 
+func expandCustomDomainActivatedResourceArray(input []interface{}) *[]cdn.ActivatedResourceReference {
+	results := make([]cdn.ActivatedResourceReference, 0)
+
+	// NOTE: I have confirmed with the service team that this is required to be an explicit "nil" value, an empty
+	// list will not work. I had to modify the SDK to allow for nil which in the API means disassociate the custom domains.
+	if len(input) == 0 || input == nil {
+		return nil
+	}
+
+	// Normalize these values, if these are imported from portal the will all be lowercased...
```

Comment:
this can be fixed in the Read function - the Expand pulls from the config, the Flatten pulls from the API data

Reply:
Fixed.